microsoft/testsuites/core/lsvmbus.py

```diff
@@ -104,3 +108,127 @@ def lsvmbus_count_devices_channels(self, node: Node) -> None:
+        device_path = f"/sys/bus/vmbus/devices/{vmbus_heartbeat_device.device_id}"
+        if not ls.path_exists(device_path):
+            raise LisaException(f"{device_path} doesn't exist")
+        vmbus_driver_files: List[str] = [
+            "channel_vp_mapping",
+            "class_id",
+            "device",
+            "device_id",
+            "id",
+            "in_intr_mask",
+            "in_read_bytes_avail",
+            "in_read_index",
+            "in_write_bytes_avail",
+            "in_write_index",
+            "modalias",
+            "out_intr_mask",
+            "out_read_bytes_avail",
+            "out_read_index",
+            "out_write_bytes_avail",
+            "out_write_index",
+            "state",
+            "uevent",
+            "vendor",
+        ]
+        vmbus_channel_files: List[str] = [
+            "cpu",
+            "events",
+            "in_mask",
+            "interrupts",
+            "out_mask",
+            "read_avail",
+            "write_avail",
+        ]
+        in_out_files: List[str] = [
+            "intr_mask",
+            "read_bytes_avail",
+            "write_bytes_avail",
+        ]
+        # driver_override file not existing - contain recent commits
+        if not ls.path_exists(f"{device_path}/driver_override"):
+            vmbus_driver_files.append(
+                [
+                    "monitor_id",
+                    "server_monitor_conn_id",
+                    "server_monitor_latency",
+                    "server_monitor_pending",
+                    "client_monitor_conn_id",
+                    "client_monitor_latency",
+                    "client_monitor_pending",
+                ]
+            )
+            vmbus_channel_files.append(["latency", "pending"])
+        for file in vmbus_driver_files:
+            if not ls.path_exists(f"{device_path}/{file}"):
+                raise LisaException(f"{device_path}/{file} doesn't exist")
+        # /sys/bus/vmbus/devices/fd149e91-82e0-4a7d-afa6-2a4166cbd7c0/channels/8
+        channels = ls.list_dir(f"{device_path}/channels/")
+        for file in vmbus_channel_files:
+            for channel in channels:
+                if not ls.path_exists(f"{channel}/{file}"):
+                    raise LisaException(f"{channel}/{file} doesn't exist")
+        for in_out_file in in_out_files:
+            in_file_value = node.tools[Cat].read(
+                f"{device_path}/in_{in_out_file}", force_run=True, sudo=True
+            )
+            out_file_value = node.tools[Cat].read(
+                f"{device_path}/out_{in_out_file}", force_run=True, sudo=True
+            )
+            assert_that(in_file_value).described_as(
+                f"value of {device_path}/in_{in_out_file} is not equal to"
+                f" {device_path}/out_{in_out_file}"
+            ).is_equal_to(out_file_value)
+        for channel in channels:
+            origin_interrupts = node.tools[Cat].read(
```

Comment:
Get original values for all channels before the loop. So it gives more time to other channels to increase. So the overall waiting time should be much shorter.